janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLStoreManager.java

```diff
@@ -189,99 +189,101 @@ public CQLStoreManager(final Configuration configuration) throws BackendExceptio
-                }
-
-                final SSLContext sslContext = SSLContext.getInstance("TLS");
-                sslContext.init(null, trustManagers, null);
+            }
 
-                final JdkSSLOptions sslOptions = JdkSSLOptions.builder()
+            if (configuration.get(SSL_ENABLED)) {
+                try {
+                    final TrustManager[] trustManagers;
+                    try (final FileInputStream keyStoreStream = new FileInputStream(configuration.get(SSL_TRUSTSTORE_LOCATION))) {
+                        final KeyStore keystore = KeyStore.getInstance("jks");
+                        keystore.load(keyStoreStream, configuration.get(SSL_TRUSTSTORE_PASSWORD).toCharArray());
+                        final TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+                        trustManagerFactory.init(keystore);
+                        trustManagers = trustManagerFactory.getTrustManagers();
+                    }
+
+                    final SSLContext sslContext = SSLContext.getInstance("TLS");
+                    sslContext.init(null, trustManagers, null);
+
+                    final JdkSSLOptions sslOptions = JdkSSLOptions.builder()
                         .withSSLContext(sslContext)
                         .build();
-                builder.withSSL(sslOptions);
+                    builder.withSSL(sslOptions);
 
-            } catch (NoSuchAlgorithmException | CertificateException | IOException | KeyStoreException | KeyManagementException e) {
-                throw new PermanentBackendException("Error initialising SSL connection properties", e);
+                } catch (NoSuchAlgorithmException | CertificateException | IOException | KeyStoreException | KeyManagementException e) {
+                    throw new PermanentBackendException("Error initialising SSL connection properties", e);
+                }
             }
+            cluster = builder.build();
         }
-
-        return builder.build();
     }
 
-    private Session initializeSession(final String keyspaceName) {
-        final Configuration configuration = getStorageConfig();
-        final Map<String, Object> replication = Match(configuration.get(REPLICATION_STRATEGY)).of(
-                Case($("SimpleStrategy"), strategy -> HashMap.<String, Object> of("class", strategy, "replication_factor", configuration.get(REPLICATION_FACTOR))),
+    private synchronized void initializeSession(final String keyspaceName) {
+        if (!sessions.containsKey(keyspaceName)) {
+            sessions.put(keyspaceName, cluster.connect());
+            final Configuration configuration = getStorageConfig();
+            final Map<String, Object> replication = Match(configuration.get(REPLICATION_STRATEGY)).of(
+                Case($("SimpleStrategy"), strategy -> HashMap.<String, Object>of("class", strategy, "replication_factor", configuration.get(REPLICATION_FACTOR))),
```

Comment:
@amcp These are showing as changes but it's just whitespace (indented under the cluster/session==null check). This is original code from #166.